### PR TITLE
Better type checking with phpstan

### PR DESCRIPTION
**Pull request by codedge**
This PR introduces phpstan for better type checks - for the `src/` and for the `test/` files.

### CODE REVIEW

File: tests/Endpoints/SearchTest.php [hunk 1]
```diff
@@ -497,11 +497,11 @@ public function testBasicSearchWithTransformFacetsDritributionOptionToFilter():
...
         $this->index->waitForPendingUpdate($response['updateId']);
 
-        $filterAllFacets = function (array $facets) {
-            $filterOneFacet = function (array $facet) {
+        $filterAllFacets = function (array $facets): array {
+            $filterOneFacet = function (array $facet): array {
                 return array_filter(
                     $facet,
-                    function ($v, $k) { return $v > 1; },
+                    function ($v, $k): bool { return $v > 1; },
```
Comment (bidoubiwa) [inline tests/Endpoints/SearchTest.php:504]:
```suggestion
                    function (int $facetValue): bool { return $facetValue > 1; },
```

**Review by bidoubiwa — COMMENTED** [review]
Thanks so much for this PR!

I have tried the code locally, but I couldn't make the linter fail. I tried to put wrong types  in some functions for example: 

```php
    public function create(string $uid, int $options = []): self
```

But the linting was successful! Do you have an idea in which situation it fails? I also realize `composer lint` did not work so 'im going to do a PR on this issue!

File: tests/Endpoints/ClientTest.php [hunk 2]
```diff
@@ -67,7 +67,7 @@ public function testGetAllIndexes(): void
         $this->assertIsArray($response);
         $this->assertCount(2, $response);
 
-        $uids = array_map(function ($index) {
+        $uids = array_map(function ($index): ?string {
```
[the remark this answers is not in the document]
Comment (codedge) [reply tests/Endpoints/ClientTest.php:70]: Well, best is to define strong types, wherever possible. If a variable should represent a string, the define it as `string`. This is not always possible of course, as you depend on 3rd-party software.

Regarding the change:
The `getUid` returns either a `string` or `null` - see the [method declaration](https://github.com/meilisearch/meilisearch-php/blob/main/src/Endpoints/Indexes.php#L71). So a return of `?string` is the correct return type here.

If you think this is wrong, I suggest you create separate PR and change:

- the `uid` property in `Entrypoint/Indexes.php` to a string
- the `getUid()` method to return always string
- All the calls to this method to use only a string

File: src/Http/Client.php [hunk 3]
```diff
@@ -175,7 +175,7 @@ private function execute(RequestInterface $request)
 
     private function buildQueryString(array $queryParams = []): string
     {
-        return $queryParams ? '?'.http_build_query($queryParams) : '';
+        return \count($queryParams) > 0 ? '?'.http_build_query($queryParams) : '';
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply src/Http/Client.php:178]: Thanks a lot for this information

File: tests/Endpoints/ClientTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply tests/Endpoints/ClientTest.php:70]: Thanks a lot, I will look into it but you are right with your code!

**Comment by bidoubiwa** [comment]
Did you find out how to break linting tests?

**Comment by codedge** [comment]
I am unsure how you mean this: Isn't it the goal to make it not break?
The tests run fine and the static code analysis (phpstan) also does not return any issues with the code... at least not for the configured _level 1_.

You can just [increase the level to 2](https://github.com/meilisearch/meilisearch-php/pull/169/files#diff-0361f0c81f363476ddc6f44ab36fcbe66ee685d5f4c2a46b054924591544b766) or more and you fill find lots of errors, or things that are not clearly configured and annotated, so that the static code analysis can correctly determine it.

**Review by bidoubiwa — APPROVED** [review]
Hey! I see, I mixed up phpstan and php-cs-fixer. LGTM 🔥 Thanks a lot

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**